xarray/core/duck_array_ops.py

```diff
@@ -236,6 +236,16 @@ def astype(data, dtype, **kwargs):
     return data.astype(dtype, **kwargs)
 
 
+def view(data, *args, **kwargs):
+    if hasattr(data, "__array_namespace__"):
+        xp = get_array_namespace(data)
+        if xp == np:
+            # numpy currently doesn't have a view:
+            return data.view(*args, **kwargs)
+        return xp.view(data, *args, **kwargs)
```

Comment:
Which package supports this? I can't find xp.view in the standard https://data-apis.org/array-api/draft/API_specification/index.html

Reply:
Oh, I didn't realize this was implementing a standard when @kmuehlbauer brought it up in #9815. As in #9815, it works for NumPy and Dask. I don't know about any other packages other than tests don't seem to fail about it specifically.

Reply:
I didn't check this fully, but to be honest, I also tried it the same way locally.

@Illviljan can you suggest a way forward here, so that it is usable for numpy and dask?

Reply:
@dcherian Could you have a look here, too? This would unblock @QuLogic for some Fedora builds. Is the implementation of view the right way?

Reply:
Using `hasattr(__array_namespace__)` implies the method is included in the array api standard. But the standard does not include a `.view.` That's why I simply don't think the `__array_namespace__` should be used in this case. 

But using `.astype(copy=False)` does indeed seem like a nice way forward, no?

Reply:
view and astype are very different things no? view simply reinterprets the bytes as a different type, astype preserves the value and may change number of bytes to do so, for example. I'm not really sure what to do here. Perhaps just xfail? Or else you'll have to track down what netcdf-java does, which is the ultimate source of this _Unsigned convention IIRC

Reply:
Just checking in here again to see if you had an idea which way to move forward.

Reply:
Sorry @QuLogic for the delay. I'm not sure what's the best way forward. I've consulted GDAL's netcdf driver and they use some construct like `static_cast<int>`/`static_cast<uint>`.

Reply:
@QuLogic According to the suggestions of @Illviljan and @dcherian (and while looking again https://github.com/numpy/numpy/issues/22951) I've pushed a fix which I believe might fix the issue, or at least avoid the undefined cast. Would you mind checking this on your workflows? Thanks!

If this works we could clean up the PR and go ahead and merge.